### PR TITLE
Fix for Laravel 9 lang path

### PR DESCRIPTION
**Pull request by promatik**
Fix for #85.
This time I tested with a proper Laravel 9 installation, not our Demo that's still on Laravel 8 😅

### CODE REVIEW

**Comment by pxpm** [comment]
If we do the change like this we need to tag a new major since `lang_path` only exists in L9.

I would do it just to avoid future conflicts like in backup. 👍

**Comment by tabacitu** [comment]
> If we do the change like this we need to tag a new major since lang_path only exists in L9.

Ouch! Good catch @pxpm . I agree with the spirit, but that would make us maintain two versions of this:
- v4 for Backpack v5 and L8
- v5 for Backpack v5 and L9

Can't we fix this with a conditional somehow, to make v4 work for both L8 and L9? Both if you use the _old_ lang location and if you're using the _new_ lang location? We can then remove that conditional when we drop L8 support next year.

**Comment by pxpm** [comment]
Hey @tabacitu  I am sure we can, but do we want to keep that rabbit hole open and running? 

V4 should support B4 and L8 .. V5 should be b5 and l9. 

Supporting that array of versions in crud is already a though job, having it in secondary packages seems to me just additional wood for the fireplace. 

Anyway, this is more a business perspective issue, we as developers are ready for the storm, so . Your business, your rules. 👍

**Comment by tabacitu** [comment]
> V4 should support B4 and L8 .. V5 should be b5 and l9.

Backpack v5 also supports Laravel 8 - that's my point, we should not forget that. And to make it easier for us maintainers, we should make our addons depend on the Backpack version, not the Laravel version (whenever possible).

**Comment by pxpm** [comment]
Indeed, that's my point too. I still think that V5 should have followed the L9 and dropped support for L8. This is just my opinion, I agree on why we did it there and it makes sense, but do we need it for all the packages especially this "secondary" ones? Can't we just say: use V4 for Laravel 8 ( no matter the crud version) and V5 for L9, "we recommend you to update as we are now only pushing new features for the V5. 

We don't let anyone behing, what worked still work. L9 is an LTS, no worries pushing support for it. 

Nice rumbling to start the morning 🌅🌄🌅🌄 hehe

**Comment by tabacitu** [comment]
Supporting just one Laravel version sure would be easier for us. 

But it wouldn't be as easy for devs to upgrade. Now they can:
- upgrade to Backpack v5, see everything works fine;
- then upgrade to Laravel v9, if something's wrong it's Laravel's fault, not Backpack;

We'll consider supporting just one version of Laravel next time around, I promise. Until then... the hard truth is that we should support both L8 and L9 in addons when we can, that makes them easier to maintain.

**Comment by pxpm** [comment]
Ok.. it's easier to make it backwards compatible here since a helper for the `lang_path` in helpers.php would solve it like we solved the Countable in previous versions of PHP.

**Review by pxpm — APPROVED** [review]
@promatik I've just merged here a backwards compatible solution for L8. Also fixed the config merging as it was not working before. You needed to publish the config file for it to work. 

I've tested in L8 and L9.

File: src/app/Services/LangFiles.php [hunk 1]
```diff
@@ -220,12 +228,12 @@ private function setArrayValue(&$data, $keys, $value)
...
     private function getFilePath()
     {
-        return base_path("resources/lang/{$this->lang}/{$this->file}.php");
+        return $this->getLangPath().'/'.$this->file.'.php';
     }
 
     private function getLangPath()
     {
-        return base_path("resources/lang/{$this->lang}/");
+        return app()->langPath().'/'.$this->lang;
```
Comment (promatik) [inline src/app/Services/LangFiles.php:236]: @pxpm does `app()->langPath()` work on Laravel 8?

Comment (promatik) [inline src/app/Services/LangFiles.php:236]: @pxpm does `app()->langPath()` work on Laravel 8?

Comment (pxpm) [reply src/app/Services/LangFiles.php:236]: Yes. 👍 Difference is that in L9 it accepts the `$path` parameter.

Comment (pxpm) [reply src/app/Services/LangFiles.php:236]: Answered in another. But yes is the answer 👍

**Comment by tabacitu** [comment]
@promatik triple-check please? Thumbs up?

**Comment by promatik** [comment]
Tested, and it's working 👌

I was about to make a suggestion to improve a little bit the code, but it wouldn't be perfect anyway, because of the L8 support. So, in the next version we move to the proper/clean solution.

And thanks Laravel for being always moving stuff from one place to another 😅

**Pull request merged**